test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,160 @@
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [1.])
+
+    def testDuplicatePeaks(self):
+        # Similar test but for duplicte peaks.
+        # "Mirrors" test3DuplicatePeaks, test case in PitchSalienceFunction.
+        freq_speaks = [55, 55, 55]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [3.])
+
+    def testRegression3Peaks(self):
+        # Provide 3 input peaks with a unit magnitude and
+        # validate that the output salience with previously calculated values
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [ 1., 103., 315., 195., 125.,  75.,  37.] )
+        self.assertAlmostEqualVectorFixedPrecision(values, [1.1899976, 1., 1., 0.8, 0.64000005, 0.512, 0.40960002], 6)
+
+    def testRegression3PeaksHighBinResolution(self):
+        # Checks on config. values binResolution being too high for the input in question
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction(binResolution=2400)(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(values, [])
+
+    def testRegressionBadMaxFreq(self):
+        # Checks on config. values maxFrequency being too low for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(maxFrequency=50)(calculatedPitchSalience))
+
+    def testRegressionBadMinFreq(self):
+        # Checks on config. values minFrequency being too high for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(minFrequency=20000)(calculatedPitchSalience))
+
+    def testRegressionBadRefFreq(self):
+        # Checks on config. values referenceFrequency being too high for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(referenceFrequency=20000)(calculatedPitchSalience))
```

Comment:
Which algorithm is generating this error? PeakDetection of PitchSalienceFunction? We  can catch this issue in the corresponding test instead of here.

Reply:
PitchSalienceFunctionPeaks generates the error

Reply:
PitchSalienceFunction is used to prepare the input for psfp.
it is not throwing the error.